core/src/main/java/hudson/model/Computer.java

```diff
@@ -360,7 +355,8 @@ public AnnotatedLargeText<Computer> getLogText() {
      */
     @Exported
     public OfflineCause getOfflineCause() {
-        return offlineCause;
+        var temporaryOfflineCause = getNodeOrDie().getTemporaryOfflineCause();
+        return temporaryOfflineCause == null ? offlineCause : temporaryOfflineCause;
```

Comment:
Retaining previous behaviour : if a temporary offline cause is defined, from outside PoV, it replaces the technical offline cause.